### PR TITLE
Add danbooru extension

### PR DESCRIPTION
**Pull request by abelroinsard**
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/danbooru/src/components/danbooruList.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+
+      const posts: PostProps[] = (await response.json()) as PostProps[];
+      setPosts(posts);
+    } catch (error) {
+      if (error instanceof Error) {
+        showToast({ title: "Error", message: error.message + ". Did you correctly set up your API Key ?" });
+      } else {
+        showToast({ title: "Error", message: "An unknown error occurred" });
```
Comment (pernielsentikaer) [inline extensions/danbooru/src/components/danbooruList.tsx:37]: Could we use [failure style](https://developers.raycast.com/api-reference/feedback/toast#toast.style) when something goes wrong?

File: extensions/danbooru/package.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          "name": "username",
+          "description": "The name of your Danbooru account. Found above 'statistics' on your account page. Not your user ID!",
+          "type": "textfield",
+          "required": true,
+          "title": "Danbooru Username",
+          "placeholder": "Username"
+        }, {
+          "name": "apiKey",
+          "description": "An API key is needed in order to perform requests. It is found in your Danbooru profile page. This extensions needs at least the following permissions: posts:index, posts:show and posts:show_seq",
+          "type": "textfield",
```
Comment (pernielsentikaer) [inline extensions/danbooru/package.json:32]:
```suggestion
          "type": "password",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/danbooru/src/components/danbooruList.tsx (hunk 1 above)
Comment (kaitomomotta) [reply extensions/danbooru/src/components/danbooruList.tsx]: added in 58d8a571c3f8a59ef93e56544237d6c4d1484cfe. thanks for pointing it out!

File: extensions/danbooru/package.json (hunk 2 above)
Comment (kaitomomotta) [reply extensions/danbooru/package.json]: added in 58d8a571c3f8a59ef93e56544237d6c4d1484cfe. Is the difference with `textfield` related to encryption or something ? I did not really notice anything different in my testing.

Comment (pernielsentikaer) [reply extensions/danbooru/package.json]: Yes it is 🙂

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**